deploy/openshift/docs/Openshift-KVM.md

```diff
@@ -0,0 +1,380 @@
+
+The following procedure works also if bare metal servers are used. If there are existing DNS, DHCP, HTTP, PXE servers, update services following examples [here](bare-metal-prerequisites.md) and jump to [Create Ignition Configs](Openshift-KVM.md#create-ignition-configs).
+
+The procedure follows [helper node installation guide line](https://github.com/RedHatOfficial/ocp4-helpernode/blob/master/docs/quickstart.md). Some modifications occurs when applying Contrail manifests
+
+On the hypervisor host create a working directory
+
+```
+# mkdir ~/ocp4-workingdir
+# cd ~/ocp4-workingdir
+```
+
+### Create a virtual network
+
+Download the virtual network configuration file, virt-net.xml
+```
+# wget https://raw.githubusercontent.com/RedHatOfficial/ocp4-helpernode/master/docs/examples/virt-net.xml
+```
+Create a virtual network using this file file provided in this repo (modify if you need to).
+
+```
+# virsh net-define --file virt-net.xml
+```
+Set it to autostart on boot
+
+```
+# virsh net-autostart openshift4
+# virsh net-start openshift4
+```
+
+### Create a CentOS 7/8 VM
+
+Download the Kickstart file for either EL 7 or EL 8 for the helper node.
+
+**EL7**
+```
+# wget https://raw.githubusercontent.com/RedHatOfficial/ocp4-helpernode/master/docs/examples/helper-ks.cfg -O helper-ks.cfg
+```
+
+**EL 8**
+```
+# wget https://raw.githubusercontent.com/RedHatOfficial/ocp4-helpernode/master/docs/examples/helper-ks8.cfg -O helper-ks.cfg
+```
+Edit `helper-ks.cfg` for your environment and use it to install the helper. The following command installs it "unattended".
+
+**EL7**
+```
+# virt-install --name="ocp4-aHelper" --vcpus=2 --ram=4096 \
+--disk path=/var/lib/libvirt/images/ocp4-aHelper.qcow2,bus=virtio,size=30 \
+--os-variant centos7.0 --network network=openshift4,model=virtio \
+--boot hd,menu=on --location /var/lib/libvirt/iso/CentOS-7-x86_64-Minimal-2003.iso \
+--initrd-inject helper-ks.cfg --extra-args "inst.ks=file:/helper-ks.cfg" --noautoconsole
+```
+
+**EL 8**
+```
+# virt-install --name="ocp4-aHelper" --vcpus=2 --ram=4096 \
+--disk path=/var/lib/libvirt/images/ocp4-aHelper.qcow2,bus=virtio,size=50 \
+--os-variant centos8 --network network=openshift4,model=virtio \
+--boot hd,menu=on --location /var/lib/libvirt/iso/CentOS-8.2.2004-x86_64-dvd1.iso \
+--initrd-inject helper-ks.cfg --extra-args "inst.ks=file:/helper-ks.cfg" --noautoconsole
+```
+
+The provided Kickstart file installs the helper with the following settings (which is based on the virt-net.xml file that was used before).
+
+- IP - 192.168.7.77
+- NetMask - 255.255.255.0
+- Default Gateway - 192.168.7.1
+- DNS Server - 8.8.8.8
+
+You can watch the progress by lauching the viewer
+```
+# virt-viewer --domain-name ocp4-aHelper
+```
+
+Once it's done, it'll shut off...turn it on with the following command
+```
+# virsh start ocp4-aHelper
+```
+
+### Prepare the Helper Node
+
+After the helper node is installed; login to it
+```
+# ssh -l root 192.168.7.77
```

Comment:
paste IP here instead if it was parametrized before,  it can even be renamed to HELPER_IP